### PR TITLE
Bugfix/427 ts bounding

### PR DESCRIPTION
**Pull request by rma-rripken**
#427

### CODE REVIEW

File: cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java [hunk 1]
```diff
@@ -1,8 +1,7 @@
 package cwms.cda.data.dao;
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java:1]: Nice organization. We should start slowing working this into the other DAOs

File: cwms-data-api/src/test/java/cwms/cda/api/CatalogControllerTestIT.java [hunk 2]
```diff
@@ -123,7 +150,124 @@ void test_all_office_pagination_works() {
...
+                .body("entries[1].name",equalTo("Pine Flat-Outflow.Stage.Inst.15Minutes.0.two"))
+                .body("entries[2].name",equalTo("Wet Meadows.Depth-SWE.Inst.15Minutes.0.four"))
+                .body("entries[3].name",equalTo("Wet Meadows.Depth-SWE.Inst.15Minutes.0.two"))
+                ;
+    }
+
+    @Test
+    void test_loc_group() {
+
+        String aToM = "A to M";
```
Comment (MikeNeilson) [inline cwms-data-api/src/test/java/cwms/cda/api/CatalogControllerTestIT.java]: should be final.

File: cwms-data-api/src/test/java/cwms/cda/api/CatalogControllerTestIT.java [hunk 3]
```diff
@@ -123,7 +150,124 @@ void test_all_office_pagination_works() {
...
+                .statusCode(is(200))
+                .body("$", hasKey("total"))
+                .body("total", is(2))
+                .body("$", hasKey("entries"))
+                .body("entries.size()",is(2))
+                .body("entries[0].name",equalTo("Alder Springs.Precip-Cumulative.Inst.15Minutes.0.raw-cda"))
+                .body("entries[1].name",equalTo("Alder Springs.Precip-INC.Total.15Minutes.15Minutes.calc-cda"))
+        ;
+
+        String nToZ = "N to Z";
```
Comment (MikeNeilson) [inline cwms-data-api/src/test/java/cwms/cda/api/CatalogControllerTestIT.java]: should be final.

**Review by MikeNeilson — CHANGES_REQUESTED** [review]
I principal I found one minor nitpick, in two locations.

File: cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java (hunk 1 above)
Comment (rma-rripken) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java:1]: Thanks.   Future change might be to make parameter object part of the method signatures in the interfaces and drop the methods with the huge number of parameters. Hopefully adding more options and filters will now be easier.

File: cwms-data-api/src/test/java/cwms/cda/api/CatalogControllerTestIT.java (hunk 2 above)
Comment (rma-rripken) [reply cwms-data-api/src/test/java/cwms/cda/api/CatalogControllerTestIT.java]: done

File: cwms-data-api/src/test/java/cwms/cda/api/CatalogControllerTestIT.java (hunk 3 above)
Comment (rma-rripken) [reply cwms-data-api/src/test/java/cwms/cda/api/CatalogControllerTestIT.java]: done

**Review by MikeNeilson — APPROVED** [review]

**Pull request merged**